File: app/src/main/java/com/app/marvelcharacters/models/ObjectConverter.java
package com.app.marvelcharacters.models;

import androidx.room.TypeConverter;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import java.lang.reflect.Type;
import java.util.Collections;
import java.util.List;

public class ObjectConverter {

    @TypeConverter
    public static List<Character> storedStringToMyObjects(String data) {
        Gson gson = new Gson();
        if (data == null) {
            return Collections.emptyList();
        }
        Type listType = new TypeToken<List<Character>>() {}.getType();
        return gson.fromJson(data, listType);
    }

    @TypeConverter
    public static String myObjectsToStoredString(List<Character> myObjects) {
        Gson gson = new Gson();
        return gson.toJson(myObjects);
    }



    @TypeConverter
    public static List<Thumbnail> storedString(String data) {
        Gson gson = new Gson();
        if (data == null) {
            return Collections.emptyList();
        }
        Type listType = new TypeToken<List<Thumbnail>>() {}.getType();
        return gson.fromJson(data, listType);
    }

    @TypeConverter
    public static String myObjectsToStored(List<Thumbnail> myObjects) {
        Gson gson = new Gson();
        return gson.toJson(myObjects);
    }

}

File: app/src/main/java/com/app/marvelcharacters/models/ObjectConverter.kt
package com.app.marvelcharacters.models

import androidx.room.TypeConverter
import com.app.marvelcharacters.models.Results
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken

object ObjectConverter {
    @TypeConverter
    @JvmStatic
    fun stringToObject(data: String?): Results? {
        val gson = Gson()
        if (data == null) {
            return null
        }
        val listType = object : TypeToken<Results?>() {}.type
        return gson.fromJson(data, listType)
    }

    @TypeConverter
    @JvmStatic
    fun objectToString(myObjects: Results?): String {
        val gson = Gson()
        return gson.toJson(myObjects)
    }

}